scifin/marketdata/simuldata.py

```diff
@@ -173,13 +173,18 @@ def to_list(self, start_date=None, end_date=None):
         
         # Initialization
         list_ts = []
-        new_index = self.data.loc[start_date:end_date].index.to_timestamp()
+        new_index = pd.to_datetime(self.data.index[start_date:end_date])
         
         # Forming a list of timeseries
         i=0
         for c in self.data.columns:
-            tmp_df = pd.DataFrame(index=new_index, data=self.data[start_date:end_date][c].values)
-            list_ts.append(timeseries.TimeSeries(tmp_df, tz=self.tz, unit=self.units[i], name=c))
+            tmp_series = pd.Series(index=new_index, data=self.data.loc[start_date:end_date, c].values)
+            if self.units is None:
+                tmp_unit = ''
```

Comment:
What is the rationale behind setting a missing unit to an empty string?

Reply:
This is because in this function we create a list of TimeSeries. We may want to plot some of them and you can see in timeseries.simple_plot() that self.unit is used in the y-label through:
ylabel = 'Value (' + self.unit + ')'
I think this is the only reason. We could easily correct that with a condition on self.unit to generate the y-label. It would make more sense actually.